daisy_workflows/image_test/configuration/configuration-test.py

```diff
@@ -0,0 +1,349 @@
+    # tests, like os-login.
+    pass
+
+  def TestKernelCmdargs(self):
+    """
+    Ensure boot loader kernel command line args (per distro).
+    """
+    # TODO: more information needed. Follow up on:
+    # https://github.com/GoogleCloudPlatform/compute-image-tools/issues/402
+    pass
+
+  def TestHostname(self, expected_hostname):
+    """
+    Ensure hostname gets set to the instance name.
+    """
+    import socket
+    actual_hostname = socket.gethostname()
+    if expected_hostname != actual_hostname:
+      raise Exception('Hostname "%s" differs from expected "%s"' % (
+          actual_hostname, expected_hostname))
+
+  def TestRsyslogConfig(self):
+    """
+    Ensure that rsyslog is installed and configured (if the distro uses
+    rsyslog) and that the hostname is properly set in the logs on boot.
+    """
+    if not self.IsPackageInstalled('rsyslog'):
+      # rsyslog was not found, skip this test for this distro
+      return
+
+    # test if kernel and daemon messages are being logged to console. The
+    # hostname output will be checked by the step "rsyslog-hostname-test"
+    info = [
+        ['kern.info', 'RsyslogKernelConsoleTest'],
+        ['daemon.info', 'RsyslogDaemonConsoleTest'],
+    ]
+    for facility in info:
+      utils.Execute(['logger', '-p'] + facility)
+
+  def TestRootPasswordDisabled(self):
+    """
+    Ensure root password is disabled (/etc/passwd)
+    """
+    # as 'man shadow' described:
+    # If the password field contains some string that is not a valid result of
+    # crypt(3), for instance ! or *, the user will not be able to use a unix
+    # password to log in
+    #
+    # Below, not the most pythonic thing to do... but it's the easiest one
+    utils.Execute(['grep', '^root:[\!*]', '/etc/shadow'])
+
+  def TestSshdConfig(self):
+    """
+    Ensure sshd config has sane default settings
+    """
+    def parseSshdConfig(path):
+      configs = {}
+      with open(path) as f:
+        for line in filter(RemoveCommentAndStrip, f.read().split('\n')):
+          if line:
+            # use line separator for key and # values
+            entry = line.split(' ')
+            # strip dictionary value
+            configs[entry[0]] = ' '.join(entry[1:]).strip()
+      return configs
+
+    sshd_desired_configs = {
+        'PermitRootLogin': 'no',
+        'PasswordAuthentication': 'no',
+    }
+
+    actual_sshd_configs = parseSshdConfig('/etc/ssh/sshd_config')
+    for key in sshd_desired_configs:
+      if actual_sshd_configs[key] != sshd_desired_configs[key]:
+        raise Exception('Sshd key "%s" should be "%s" and not "%s"' % (
+            key, sshd_desired_configs[key], actual_sshd_configs[key]))
+
+  def TestPackageManagerConfig(self):
+    """
+    Ensure apt/yum repos are setup for GCE repos.
+    """
+    # The google-cloud-sdk package should only be found if GCE repo is setup
+    self.EnsurePackageIsInstalled('google-cloud-sdk')
```

Comment:
actually I see that this is not the best way of verifying it... (rhel-6 has gcloud/gsutil binaries installed by other means than rpm). 
I'll rewrite this check to really check if `packages.cloud.google.com` is used by apt/yum

Reply:
done